api/modules/publishedFiles/model.js

```diff
@@ -1,9 +1,101 @@
 var BaseModel = require('../../classes/base_model');
 
+var File = require('../files/model');
+
 var instanceProps = {
   tableName: 'publishedFiles',
   project: function() {
     return this.belongsTo(require('../publishedProjects/model'));
+  },
+  queries: function() {
+    var self = this;
+    var FilesTable = File.prototype;
+    var PublishedFile = this.constructor;
+
+    return {
+      getOne: function(id) {
+        return new PublishedFile().query()
+        .where(self.column('id'), id)
+        .then(function(publishedFiles) {
+          return publishedFiles[0];
+        });
+      },
+      getAllNewFiles: function(projectId) {
+        return new File().query()
+        .leftOuterJoin(self.tableName,
+                       FilesTable.column('id'),
+                       self.column('file_id'))
+        .where(FilesTable.column('project_id'), projectId)
+        .whereNull(self.column('file_id'))
```

Comment:
Or is this saying, "return the file record where no publishedFile references it"?